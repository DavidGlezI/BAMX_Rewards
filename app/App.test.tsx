import React from "react";
import renderer from "react-test-renderer";
import RootLayout from "./_layout";
import LoginScreen from "./login";
import RegisterScreen from "./register";
import TabHomeScreen from "./(tabs)/home";
import TabDonateScreen from "./(tabs)/donate";
import TabPromotionsScreen from "./(tabs)/promotions";
import axios from "axios";
import * as api from "../util/useApi";
import { act, render, waitFor } from "@testing-library/react-native";
import TabAccountScreen from "./(tabs)/account";

jest.mock("axios");
const mockedAxios = axios as jest.Mocked<typeof axios>;

describe("Snapshot tests for all pages", () => {
  beforeEach(() => jest.clearAllMocks());

  test("Login page renders correctly", () => {
    const tree = renderer.create(<LoginScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Register page renders correctly", () => {
    const tree = renderer.create(<RegisterScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Tabs donate page renders correctly", () => {
    const tree = renderer.create(<TabDonateScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Tabs promotions page renders correctly and makes api call", async () => {
    const spy = jest.spyOn(api, "useFetch");
    let tree;
    await act(async () => {
      tree = await renderer.create(<TabPromotionsScreen />);
    });
    expect(tree).toMatchSnapshot();
    // get data from backend on render
    expect(spy).toHaveBeenCalledWith("restaurants");
  });

  test("Tabs home page renders correctly and makes api call", async () => {
    const spy = jest.spyOn(api, "useFetch");
    let tree;
    await act(async () => {
      tree = await renderer.create(<TabHomeScreen />);
    });
    expect(tree).toMatchSnapshot();
    // get data from backend on render
    expect(spy).toHaveBeenCalledWith("restaurants");
  });

  test("Tabs account page renders correctly and makes api call", async () => {
    const spy = jest.spyOn(api, "useFetch");
    let tree;
    await act(async () => {
      tree = await renderer.create(<TabAccountScreen />);
    });
    expect(tree).toMatchSnapshot();
    // get data from backend on render
    expect(spy).toHaveBeenCalledWith("donations");
    expect(spy).toHaveBeenCalledWith("user");
  });
});
