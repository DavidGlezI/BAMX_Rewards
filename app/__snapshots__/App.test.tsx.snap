// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests for all pages Login page renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#efeff4",
      },
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-end",
        "marginBottom": 30,
        "position": "relative",
      },
    ]
  }
>
  <Image
    source={1}
    style={
      {
        "height": 808.4848484848485,
        "minWidth": 800,
        "overflow": "hidden",
        "position": "absolute",
        "top": 0,
      }
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "#efeff4",
        },
        {
          "backgroundColor": "#fff",
          "borderRadius": 40,
          "flex": 0.5,
          "padding": 40,
          "shadowColor": "black",
          "shadowOffset": {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 8,
          "width": "90%",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#000",
          },
          {
            "fontSize": 24,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Iniciar sesión
    </Text>
    <Text
      onPress={[Function]}
      style={
        [
          {
            "color": "#000",
          },
          {
            "fontSize": 14,
            "marginBottom": 60,
            "marginTop": 10,
          },
        ]
      }
    >
      o Únete a BAMX
    </Text>
    <TextInput
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Correo Electrónico"
      placeholderTextColor="gray"
      style={
        [
          {
            "borderBottomWidth": 1,
            "height": 40,
            "marginBottom": 12,
            "paddingLeft": 8,
          },
          {
            "borderColor": "#CCCCCC",
          },
        ]
      }
      value=""
    />
    <TextInput
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Contraseña"
      placeholderTextColor="gray"
      secureTextEntry={true}
      style={
        [
          {
            "borderBottomWidth": 1,
            "height": 40,
            "marginBottom": 12,
            "paddingLeft": 8,
          },
          {
            "borderColor": "#CCCCCC",
          },
        ]
      }
      value=""
    />
    <View
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "#E4022E",
          "borderRadius": 40,
          "height": 55,
          "justifyContent": "center",
          "marginTop": 20,
          "width": 240,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#000",
            },
            {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "700",
              "textAlign": "center",
            },
          ]
        }
      >
        Iniciar Sesión
      </Text>
    </View>
  </View>
</View>
`;

exports[`Snapshot tests for all pages Register page renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#efeff4",
      },
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-end",
        "marginBottom": 30,
        "position": "relative",
      },
    ]
  }
>
  <Image
    source={1}
    style={
      {
        "height": 808.4848484848485,
        "minWidth": 800,
        "overflow": "hidden",
        "position": "absolute",
        "top": 0,
      }
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "#efeff4",
        },
        {
          "backgroundColor": "#fff",
          "borderRadius": 40,
          "flex": 0.5,
          "padding": 40,
          "shadowColor": "black",
          "shadowOffset": {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 8,
          "width": "90%",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#000",
          },
          {
            "fontSize": 24,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Registrate
    </Text>
    <Text
      onPress={[Function]}
      style={
        [
          {
            "color": "#000",
          },
          {
            "fontSize": 14,
            "marginBottom": 30,
            "marginTop": 10,
          },
        ]
      }
    >
      o Iniciar sesión
    </Text>
    <TextInput
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Nombre y Apellido"
      placeholderTextColor="gray"
      style={
        [
          {
            "borderBottomWidth": 1,
            "height": 40,
            "marginBottom": 12,
            "paddingLeft": 8,
          },
          {
            "borderColor": "#CCCCCC",
          },
        ]
      }
      value=""
    />
    <TextInput
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Correo Electrónico"
      placeholderTextColor="gray"
      style={
        [
          {
            "borderBottomWidth": 1,
            "height": 40,
            "marginBottom": 12,
            "paddingLeft": 8,
          },
          {
            "borderColor": "#CCCCCC",
          },
        ]
      }
      value=""
    />
    <TextInput
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Contraseña"
      placeholderTextColor="gray"
      secureTextEntry={true}
      style={
        [
          {
            "borderBottomWidth": 1,
            "height": 40,
            "marginBottom": 12,
            "paddingLeft": 8,
          },
          {
            "borderColor": "#CCCCCC",
          },
        ]
      }
      value=""
    />
    <View
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "#E4022E",
          "borderRadius": 40,
          "height": 55,
          "justifyContent": "center",
          "marginTop": 20,
          "width": 240,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#000",
            },
            {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "700",
              "textAlign": "center",
            },
          ]
        }
      >
        Continuar
      </Text>
    </View>
  </View>
</View>
`;

exports[`Snapshot tests for all pages Tabs account page renders correctly and makes api call 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
      "width": 750,
    }
  }
>
  <Image
    source={1}
    style={
      {
        "alignSelf": "center",
        "height": 444.6666666666667,
        "marginLeft": 20,
        "minWidth": 750,
        "overflow": "hidden",
        "position": "absolute",
        "top": 0,
        "zIndex": 0,
      }
    }
  />
  <RCTScrollView
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
  >
    <RCTRefreshControl />
    <View>
      <Text
        style={
          [
            {
              "color": "#000",
            },
            undefined,
          ]
        }
      >
        Error
      </Text>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`Snapshot tests for all pages Tabs donate page renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      {
        "alignSelf": "center",
        "height": 444.6666666666667,
        "marginLeft": 20,
        "minWidth": 750,
        "overflow": "hidden",
        "position": "absolute",
        "top": 0,
        "zIndex": 0,
      }
    }
  />
  <View
    style={
      {
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "marginTop": 120,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-evenly",
        }
      }
    >
      <RNGestureHandlerButton
        collapsable={false}
        delayLongPress={600}
        enabled={true}
        exclusive={true}
        handlerTag={1}
        handlerType="NativeViewGestureHandler"
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        touchSoundDisabled={false}
      >
        <View
          accessible={true}
          collapsable={false}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#E4022E",
              "borderRadius": 20,
              "elevation": 4,
              "flexShrink": 0,
              "height": 160,
              "justifyContent": "center",
              "margin": 10,
              "padding": 10,
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 4,
              "width": 106,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FFF",
                "borderRadius": 20,
                "elevation": 4,
                "flexShrink": 0,
                "height": 160,
                "justifyContent": "center",
                "margin": 10,
                "padding": 10,
                "shadowColor": "rgba(0, 0, 0, 0.1)",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 4,
                "width": 106,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#000",
                  "fontSize": 12,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Basico
            </Text>
            <Text
              style={
                {
                  "color": "#E6012E",
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              20
              $
            </Text>
            <Text
              style={
                {
                  "color": "#000",
                  "fontSize": 12,
                  "textAlign": "center",
                }
              }
            >
              Plan Mensual
            </Text>
            <Text
              style={
                {
                  "fontSize": 8,
                  "fontWeight": "400",
                }
              }
            >
              recibes
            </Text>
            <Text
              style={
                {
                  "color": "red",
                  "display": "flex",
                  "fontSize": 8,
                  "textAlign": "center",
                }
              }
            >
              40
               Puntos
            </Text>
          </View>
        </View>
      </RNGestureHandlerButton>
      <RNGestureHandlerButton
        collapsable={false}
        delayLongPress={600}
        enabled={true}
        exclusive={true}
        handlerTag={2}
        handlerType="NativeViewGestureHandler"
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        touchSoundDisabled={false}
      >
        <View
          accessible={true}
          collapsable={false}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#E4022E",
              "borderRadius": 20,
              "elevation": 4,
              "flexShrink": 0,
              "height": 160,
              "justifyContent": "center",
              "margin": 10,
              "padding": 10,
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 4,
              "width": 106,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FFF",
                "borderRadius": 20,
                "elevation": 4,
                "flexShrink": 0,
                "height": 160,
                "justifyContent": "center",
                "margin": 10,
                "padding": 10,
                "shadowColor": "rgba(0, 0, 0, 0.1)",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 4,
                "width": 106,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#000",
                  "fontSize": 12,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Medio
            </Text>
            <Text
              style={
                {
                  "color": "#E6012E",
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              50
              $
            </Text>
            <Text
              style={
                {
                  "color": "#000",
                  "fontSize": 12,
                  "textAlign": "center",
                }
              }
            >
              Plan Mensual
            </Text>
            <Text
              style={
                {
                  "fontSize": 8,
                  "fontWeight": "400",
                }
              }
            >
              recibes
            </Text>
            <Text
              style={
                {
                  "color": "red",
                  "display": "flex",
                  "fontSize": 8,
                  "textAlign": "center",
                }
              }
            >
              100
               Puntos
            </Text>
          </View>
        </View>
      </RNGestureHandlerButton>
      <RNGestureHandlerButton
        collapsable={false}
        delayLongPress={600}
        enabled={true}
        exclusive={true}
        handlerTag={3}
        handlerType="NativeViewGestureHandler"
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        touchSoundDisabled={false}
      >
        <View
          accessible={true}
          collapsable={false}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#E4022E",
              "borderRadius": 20,
              "elevation": 4,
              "flexShrink": 0,
              "height": 160,
              "justifyContent": "center",
              "margin": 10,
              "padding": 10,
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 4,
              "width": 106,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FFF",
                "borderRadius": 20,
                "elevation": 4,
                "flexShrink": 0,
                "height": 160,
                "justifyContent": "center",
                "margin": 10,
                "padding": 10,
                "shadowColor": "rgba(0, 0, 0, 0.1)",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 4,
                "width": 106,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#000",
                  "fontSize": 12,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Avanzado
            </Text>
            <Text
              style={
                {
                  "color": "#E6012E",
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              150
              $
            </Text>
            <Text
              style={
                {
                  "color": "#000",
                  "fontSize": 12,
                  "textAlign": "center",
                }
              }
            >
              Plan Mensual
            </Text>
            <Text
              style={
                {
                  "fontSize": 8,
                  "fontWeight": "400",
                }
              }
            >
              recibes
            </Text>
            <Text
              style={
                {
                  "color": "red",
                  "display": "flex",
                  "fontSize": 8,
                  "textAlign": "center",
                }
              }
            >
              300
               Puntos
            </Text>
          </View>
        </View>
      </RNGestureHandlerButton>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-evenly",
        }
      }
    >
      <RNGestureHandlerButton
        collapsable={false}
        delayLongPress={600}
        enabled={true}
        exclusive={true}
        handlerTag={4}
        handlerType="NativeViewGestureHandler"
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        touchSoundDisabled={false}
      >
        <View
          accessible={true}
          collapsable={false}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#E4022E",
              "borderRadius": 20,
              "elevation": 4,
              "flexShrink": 0,
              "height": 160,
              "justifyContent": "center",
              "margin": 10,
              "padding": 10,
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 4,
              "width": 106,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FFF",
                "borderRadius": 20,
                "elevation": 4,
                "flexShrink": 0,
                "height": 160,
                "justifyContent": "center",
                "margin": 10,
                "padding": 10,
                "shadowColor": "rgba(0, 0, 0, 0.1)",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 4,
                "width": 106,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#000",
                  "fontSize": 12,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Plus
            </Text>
            <Text
              style={
                {
                  "color": "#E6012E",
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              300
              $
            </Text>
            <Text
              style={
                {
                  "color": "#000",
                  "fontSize": 12,
                  "textAlign": "center",
                }
              }
            >
              Plan Mensual
            </Text>
            <Text
              style={
                {
                  "fontSize": 8,
                  "fontWeight": "400",
                }
              }
            >
              recibes
            </Text>
            <Text
              style={
                {
                  "color": "red",
                  "display": "flex",
                  "fontSize": 8,
                  "textAlign": "center",
                }
              }
            >
              700
               Puntos
            </Text>
          </View>
        </View>
      </RNGestureHandlerButton>
      <RNGestureHandlerButton
        collapsable={false}
        delayLongPress={600}
        enabled={true}
        exclusive={true}
        handlerTag={5}
        handlerType="NativeViewGestureHandler"
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        touchSoundDisabled={false}
      >
        <View
          accessible={true}
          collapsable={false}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#E4022E",
              "borderRadius": 20,
              "elevation": 4,
              "flexShrink": 0,
              "height": 160,
              "justifyContent": "center",
              "margin": 10,
              "padding": 10,
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 4,
              "width": 106,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FFF",
                "borderRadius": 20,
                "elevation": 4,
                "flexShrink": 0,
                "height": 160,
                "justifyContent": "center",
                "margin": 10,
                "padding": 10,
                "shadowColor": "rgba(0, 0, 0, 0.1)",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 4,
                "width": 106,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#000",
                  "fontSize": 12,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Premium
            </Text>
            <Text
              style={
                {
                  "color": "#E6012E",
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              500
              $
            </Text>
            <Text
              style={
                {
                  "color": "#000",
                  "fontSize": 12,
                  "textAlign": "center",
                }
              }
            >
              Plan Mensual
            </Text>
            <Text
              style={
                {
                  "fontSize": 8,
                  "fontWeight": "400",
                }
              }
            >
              recibes
            </Text>
            <Text
              style={
                {
                  "color": "red",
                  "display": "flex",
                  "fontSize": 8,
                  "textAlign": "center",
                }
              }
            >
              1200
               Puntos
            </Text>
          </View>
        </View>
      </RNGestureHandlerButton>
      <RNGestureHandlerButton
        collapsable={false}
        delayLongPress={600}
        enabled={true}
        exclusive={true}
        handlerTag={6}
        handlerType="NativeViewGestureHandler"
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        touchSoundDisabled={false}
      >
        <View
          accessible={true}
          collapsable={false}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#E4022E",
              "borderRadius": 20,
              "elevation": 4,
              "flexShrink": 0,
              "height": 160,
              "justifyContent": "center",
              "margin": 10,
              "padding": 10,
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 4,
              "width": 106,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FFF",
                "borderRadius": 20,
                "elevation": 4,
                "flexShrink": 0,
                "height": 160,
                "justifyContent": "center",
                "margin": 10,
                "padding": 10,
                "shadowColor": "rgba(0, 0, 0, 0.1)",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 4,
                "width": 106,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#000",
                  "fontSize": 12,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Max
            </Text>
            <Text
              style={
                {
                  "color": "#E6012E",
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              1000
              $
            </Text>
            <Text
              style={
                {
                  "color": "#000",
                  "fontSize": 12,
                  "textAlign": "center",
                }
              }
            >
              Plan Mensual
            </Text>
            <Text
              style={
                {
                  "fontSize": 8,
                  "fontWeight": "400",
                }
              }
            >
              recibes
            </Text>
            <Text
              style={
                {
                  "color": "red",
                  "display": "flex",
                  "fontSize": 8,
                  "textAlign": "center",
                }
              }
            >
              2500
               Puntos
            </Text>
          </View>
        </View>
      </RNGestureHandlerButton>
    </View>
  </View>
  <View
    style={
      {
        "marginTop": 40,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderRadius": 20,
          "elevation": 4,
          "flexShrink": 0,
          "height": 100,
          "justifyContent": "space-evenly",
          "margin": 10,
          "padding": 10,
          "shadowColor": "rgba(0, 0, 0, 0.1)",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 4,
          "width": 335,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }
      >
        Tu donación marca la diferencia
      </Text>
      <Text
        style={
          {
            "color": "red",
            "fontSize": 12,
            "marginBottom": 6,
            "marginTop": 2,
          }
        }
      >
        ¡Únete para cambiar vidas!
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": 2,
          }
        }
      >
        <View
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderColor": "red",
              "borderRadius": 20,
              "borderWidth": 2,
              "height": 35,
              "justifyContent": "center",
              "marginLeft": 10,
              "width": 35,
            }
          }
        >
          <Text
            style={
              {
                "color": "red",
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
            5
          </Text>
        </View>
        <View
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderColor": "red",
              "borderRadius": 20,
              "borderWidth": 2,
              "height": 35,
              "justifyContent": "center",
              "marginLeft": 10,
              "width": 35,
            }
          }
        >
          <Text
            style={
              {
                "color": "red",
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
            10
          </Text>
        </View>
        <View
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderColor": "red",
              "borderRadius": 20,
              "borderWidth": 2,
              "height": 35,
              "justifyContent": "center",
              "marginLeft": 10,
              "width": 35,
            }
          }
        >
          <Text
            style={
              {
                "color": "red",
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
            15
          </Text>
        </View>
        <View
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderColor": "red",
              "borderRadius": 20,
              "borderWidth": 2,
              "height": 35,
              "justifyContent": "center",
              "marginLeft": 10,
              "width": 35,
            }
          }
        >
          <Text
            style={
              {
                "color": "red",
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
            20
          </Text>
        </View>
        <View
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderColor": "red",
              "borderRadius": 20,
              "borderWidth": 2,
              "height": 35,
              "justifyContent": "center",
              "marginLeft": 10,
              "width": 35,
            }
          }
        >
          <Text
            style={
              {
                "color": "red",
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
            50
          </Text>
        </View>
        <View
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderColor": "red",
              "borderRadius": 20,
              "borderWidth": 2,
              "height": 35,
              "justifyContent": "center",
              "marginLeft": 10,
              "width": 35,
            }
          }
        >
          <Text
            style={
              {
                "color": "red",
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
            100
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Snapshot tests for all pages Tabs home page renders correctly and makes api call 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      {
        "alignSelf": "center",
        "height": 444.6666666666667,
        "marginLeft": 20,
        "minWidth": 750,
        "overflow": "hidden",
        "position": "absolute",
        "top": 0,
        "zIndex": 0,
      }
    }
  />
  <RCTScrollView
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
  >
    <RCTRefreshControl />
    <View>
      <Text
        style={
          {
            "marginTop": 266.8,
          }
        }
      >
        Sorry there was an error loading your data
      </Text>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`Snapshot tests for all pages Tabs promotions page renders correctly and makes api call 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      {
        "alignSelf": "center",
        "height": 444.6666666666667,
        "marginLeft": 20,
        "minWidth": 750,
        "overflow": "hidden",
        "position": "absolute",
        "top": 0,
        "zIndex": 0,
      }
    }
  />
  <RCTScrollView
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
  >
    <RCTRefreshControl />
    <View>
      <Text
        style={
          {
            "marginTop": 266.8,
          }
        }
      >
        Sorry there was an error loading your data
      </Text>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
